app/src/main/java/ch/epfl/sdp/contamination/ConcreteAnalysis.java

```diff
@@ -102,27 +101,28 @@ private void modelInfectionEvolution(Map<Carrier, Integer> suspectedContacts) {
     }
 
     @Override
-    public void updateInfectionPredictions(Location location, Date startTime, Callback<Void> callback) {
-
+    public CompletableFuture<Void> updateInfectionPredictions(Location location, Date startTime) {
         Date now = new Date(System.currentTimeMillis());
-        receiver.getUserNearbyDuring(location, startTime, now, aroundMe -> {
-            modelInfectionEvolution(identifySuspectContacts(aroundMe));
-            callback.onCallback(null);
-        });
+        CompletableFuture future1 = receiver.getUserNearbyDuring(location, startTime, now)
```

Comment:
Would be useful for maintenance to give more explicit names for futures, or even directly inline them into CompletableFuture.allOf?

Reply:
Refactored